frame/identity/src/lib.rs

```diff
@@ -201,6 +202,44 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub identities: Vec<(T::AccountId, String)>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			GenesisConfig { identities: Default::default() }
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			for (account, name) in &self.identities {
+				<IdentityOf<T>>::insert(
+					account,
+					Registration {
+						info: IdentityInfo {
+							display: Data::Raw(BoundedVec::try_from(name.encode()).unwrap()),
```

Comment:
Should never .unwrap or have potential panics in the runtime, it bricks the chain

Reply:
Genesis construction is uncritical - the chain just wont ever start.

Reply:
But now it ignores the genesis config in the error case. I think panic is better.

Reply:
Ok, reverted last change